src/comm/SYCLHelpers.h

```diff
@@ -137,4 +137,30 @@ sycl_kernel_submit(
         ker);
   };
   q.submit(cgf);
-}
\ No newline at end of file
+}
+
+#define SIMD16 16
+#define SIMD32 32
+
+template <
+    typename fn_simd_16,
+    typename fn_simd_32,
+    typename range_t,
+    typename... args_t>
+static inline void sycl_kernel_submit_simd(
```

Comment:
We should not couple simd with kernel submission convention. Explicit simd width is a logic of kernel.